File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.py
import sys
def print(*args, sep=" ", end="\n"): return sys.stdout.write(sep.join(map(str, args)) + end)
def input(): return sys.stdin.readline().rstrip()

def main():
    n, m = map(int, input().split())
    board = [input() for _ in range(n)]
    min_d = 64
    for i in range(n-7):
        for j in range(m-7):
            d = 0
            for r in range(i, i+8):
                for c in range(j, j+8):
                    if (r+c)%2:
                        if board[r][c] != 'W': d += 1
                    else:
                        if board[r][c] != 'B': d += 1
            min_d = min(min_d, d, 64-d)
    print(min_d)
    return 0
if __name__ == '__main__':
    sys.exit(main())

File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.cc
#include<bits/stdc++.h>
using namespace std;
using ll = long long;

int main(){
    cin.tie(0); cout.tie(0); ios::sync_with_stdio(false);
    int n, m; cin >> n >> m;
    char board[n][m];
    for(int i = 0; i < n; i++){
        for(int j = 0; j < m; j++) cin >> board[i][j];
    }
    int min_d = 64;
    for(int i = 0; i < n-7; i++){
        for(int j = 0; j < m-7; j++){
            int d = 0;
            for(int r = i; r < i+8; r++){
                for(int c = j; c < j+8; c++){
                    if((r+c)%2){
                        if(board[r][c] != 'W') d++;
                    }
                    else{
                        if(board[r][c] != 'B') d++;
                    }
                }
            }
            min_d = min({min_d, d, 64-d}); 
        }
    }
    cout << min_d;
    return 0;
}